src/box/memtx_tx.c

```diff
@@ -365,23 +608,123 @@ memtx_tx_handle_conflict(struct txn *breaker, struct txn *victim)
 	}
 }
 
+/**
+ * Calculate size of story with its links.
+ */
+static inline size_t
+memtx_story_size(struct memtx_story *story)
+{
+	struct mempool *pool = &txm.memtx_tx_story_pool[story->index_count];
+	return pool->objsize;
+}
+
+/**
+ * Notify memory manager that a tuple referenced by @a story
+ * was replaced from primary key and that is why @a story
+ * is the only reason why the tuple cannot be deleted.
+ */
+static inline void
+memtx_tx_story_tuple_retain(struct memtx_story *story)
```

Comment:
I think the names are misleading: I'd rather it be something like
```suggestion
memtx_tx_story_track_retained_tuple(struct memtx_story *story)
```